### PR TITLE
Removed unused object

### PR DESCRIPTION
**Pull request by Jentsch**
The tests are still green. What is the purpose of those companion objects? Could I remove them? If yes I would amend this pull request with more commits.

### CODE REVIEW

**Comment by wheaties** [comment]
Take a look at the [export-hook](https://github.com/milessabin/export-hook) project. That's where they come from. To be honest, we're not getting what was hoped we'd get out of it. The idea being that not only would it help solve orphan type classes (i.e. there wouldn't have to even be a `ScalazLiftMap` but just another `object` that implemented `LiftMap` in the autolift-scalaz project.) It's also not helping with reducing import overhead (a la `reexports`) due to the path dependent types. That could change if and when someone spent some time and effort to make that project more robust in regards to type members.

TL;DR If you want to take them out, might as well take out the export-hook dependency. We might even be able to lose the macro-paradise plugin as well.

**Comment by Jentsch** [comment]
I removed all references to the export-hook, macro-paradise and scala-reflect as well. Still all tests are green.

**Comment by wheaties** [comment]
Can't argue with this change. If we're not getting the benefits we want, there's no point in using it. I think it's definitely something that should be revisited in the future.
